src/lib/components/ReactRRuleGenerator.js

```diff
@@ -20,46 +23,32 @@ class ReactRRuleGenerator extends Component {
 
       this.props.onChange(rrule);
 
-      return { data: newData, isCopied: false, rrule };
+      return { data: newData, rrule };
```

Comment:
Its the role of container to send state down to ReactRRuleGenerator via `this.props.value`.  You can convert it into local state, thats ok and I would suggest to do it within `componentWillReceiveProps`, see http://www.react.express/lifecycle_api for details.

Check source of  react-select https://github.com/JedWatson/react-select/blob/0d0ccd3acc4efdba0b32d711a6c16809a6d779b4/src/Select.js#L93